components/scream/src/share/field/field_repository.hpp

```diff
@@ -0,0 +1,76 @@
+#ifndef SCREAM_FIELD_REPOSITORY_HPP
+#define SCREAM_FIELD_REPOSITORY_HPP
+
+#include <share/scream_types.hpp>
+#include <share/error_defs.hpp>
+#include "field.hpp"
+
+#include <map>
+
+namespace scream
+{
+
+ /*
+  *  A database for all the persistent fields needed in an atm time step
+  *  We template a field repository over the memory space. The data type of the underlying view
+  *  will *always* be Real*, so that we can store all the fields in one place (regardless of
```

Comment:
Are you considering vectorization and, potentially, AD?

With vectorization, Real is OK b/c it can be the scalarized field. But, importantly for things like `reshape`, we have to be aware that the scalarized View will have padding if the pack size does not divide the number of levels. So, in summary, Real is OK here, but think through whether anything has incorrectly assumed no padding.

SFad-type AD I'm pretty sure scalarizes w/o an issue, as well. But you might check that. (I think it's reasonable to disallow DFad-type AD.)

Reply:
I'm not suggesting a chance since SFad should be fine with Real: just asking whether padding has been considered in the design.